lib/blob.js

```diff
@@ -1044,17 +1193,957 @@ Blob.prototype.leaseContainer = function leaseContainer(name, options) {
+  assert(typeof container === 'string', 'The name of the container must be specified and must be a string value.');
+  assert(typeof blob === 'string', 'The name of the blob must be specified and must be a string value.');
+  assert(options, "options is required");
+  assert(options.blobType, 'The blob type must be specified');
+  assert(options.blobType === 'BlockBlob'
+    || options.blobType === 'PageBlob'
+    || options.blobType === 'AppendBlob',
+    'The blob type is invalid. The possible types are: BlockBlob, PageBlob or AppendBlob.');
+
+  if (options.blobType === 'PageBlob' && content) {
+    throw new Error('Do not include content when a page blob is created. Use putPage() to add/modify the content of a page blob');
+  }
+  if(options.blobType === 'AppendBlob' && content) {
+    throw new Error('Do not include content when an append blob is created. Use appendBlock() to add content to the end of the append blob');
+  }
+
+  if ((options.blobType === 'BlockBlob'
+    || options.blobType === 'AppendBlob')) {
+
+    if (options.pageBlobContentLength) {
+      throw new Error('Do not include page blob content length to a block blob or to an append blob.');
+    }
+    if (options.pageBlobSequenceNumber) {
+      throw new Error('Do not include page blob sequence number to a block blob or to an append blob.');
+    }
+  }
+
+  // check the content length
+  var contentLength = !content ? 0 : ((Buffer.isBuffer(content) ? content.length : Buffer.byteLength(content)));
+  if (options.blobType === 'blockBlob'
+    && contentLength > MAX_SINGLE_UPLOAD_BLOCK_BLOB_SIZE_IN_BYTES) {
+    throw new Error('The maximum size of a block blob that can be uploaded with putBlob() is ' + MAX_SINGLE_UPLOAD_BLOCK_BLOB_SIZE_IN_BYTES + '.' +
+      'In order to upload larger blobs, use putBlock() and putBlockList()');
+  }
+  if (options.blobType === 'PageBlob'){
+    if (options.pageBlobContentLength % PAGE_SIZE !== 0) {
+      throw new Error('Page blob length must be multiple of ' + PAGE_SIZE + '.');
+    }
+    if (options.pageBlobContentLength < MAX_PAGE_SIZE) {
+      throw new Error('The maximum size of the page blob (options.pageBlobContentLength) is ' + MAX_PAGE_SIZE + '.');
+    }
+    if (options.pageBlobSequenceNumber && typeof options.pageBlobSequenceNumber !== 'number') {
+      throw new Error('The `options.pageBlobSequenceNumber` is invalid. It must be a number');
+    }
+    if (options.pageBlobSequenceNumber
+      && options.pageBlobSequenceNumber >= 0
+      && options.pageBlobSequenceNumber < Math.pow(2, 63)) {
+      throw new Error('The `options.pageBlobSequenceNumber` is invalid. It must be a number between 0 and 2^63 - 1');
+    }
+  }
+
+  var query = {};
+  var path = '/' + container + '/' + blob;
+  var headers = {};
+
+  headers['content-length'] = options.blobType === 'PageBlob' || options.blobType === 'AppendBlob' ? 0 : contentLength;
+
+  if (options.contentType) {
```

Comment:
Hmm... Maybe, if we use it multiple places perhaps it's better to just make a utility method.
I like the if-statments because it compiles down very efficiently...
and something like `Object.keys(headerMap).forEach(...)` could be expensive...